src/app/backend/file-parser/path-parser.ts

```diff
@@ -12,17 +13,138 @@ export class PathParser {
+                'The number of SegmentAppearance entries must be equal to the number of Segment entries.\n' +
+                'There are ' + segmentCPs.length + ' SegmentAppearance Entries whilst there are ' +
+                segmentPOIs.length + ' Segment entries.'
+            );
+        }
+
+        // ERROR CHECK - Ensure the structure of each SegmentAppearance entry is Correct
+        for (let i = 0; i < segmentCPs.length; i++) {
+            // Get the Segment Entry of ControlPoints
+            const segCp = segmentCPs[i].children;
+
+            if (segCp[0].tagName !== 'Id') {
+                throw new ParserError(
+                    'SegmentAppearance entry number ' + i + ' must have \"Id\" as 1st tag.'
+                );
+            }
+            const segCpId = segCp[0].firstChild.nodeValue;
+            if (segCp[1].tagName !== 'Shape') {
+                throw new ParserError(
+                    'SegmentAppearance of Id ' + segCpId + ' must have \"Shape\" as 2nd tag.'
+                );
+            }
+            const shape = segCp[1];
+            // tslint:disable-next-line:prefer-for-of
+            for (let c = 0; c < shape.children.length; c++) {
+                if (c === 1) {
+                    continue;   // Skip Control Point
+                }
+                if (shape.children[c].tagName !== 'ControlPoint') {
+                    throw new ParserError(
+                        'Shape field in SegmentAppearance of Id ' + segCpId + ' wrongly structured.\n' +
+                        'Shape field must contain a list of ControlPoint entries.'
+                    );
+                }
+                if (shape.children[c].children[0].tagName !== 'xCoordinate') {
+                    throw new ParserError(
+                        'ControlPoint entry ' + c + ' of Shape field in SegmentAppearance of Id ' + segCpId +
+                        ' wrongly formatted.\n' + 'ControlPoint 1st tag must be \"xCoordinate\".'
+                    );
+                }
+                if (shape.children[c].children[1].tagName !== 'yCoordinate') {
+                    throw new ParserError(
+                        'ControlPoint entry ' + c + ' of Shape field in SegmentAppearance of Id ' + segCpId +
+                        ' wrongly formatted.\n' + 'ControlPoint 2nd tag must be \"yCoordinate\".'
+                    );
+                }
+                if (shape.children[c].children[2].tagName !== 'Name') {
+                    throw new ParserError(
+                        'ControlPoint entry ' + c + ' of Shape field in SegmentAppearance of Id ' + segCpId +
+                        ' wrongly formatted.\n' + 'ControlPoint 3rd tag must be \"Name\".'
+                    );
+                }
+            }
+        }
+
+        // ERROR CHECK - Ensure the structure of each Segment entry is Correct
+        for (let i = 0; i < segmentPOIs.length; i++) {
+            // Get the Segment Entry of POIss
```

Comment:
POIss is used twice in this file and it appears to be a typign error (as I dont see what POIss would stand for), either be more clear on what POIss is or rename to POI's.
```suggestion
            // Get the Segment Entry of POI's
```

Reply:
Shouldn't it then be `POIs`?